src/tries.jl

```diff
@@ -82,44 +82,6 @@ end
 
 lower(c::UInt8) = UInt8('A') <= c <= UInt8('Z') ? c | 0x20 : c 
 
-"""
-    Parsers.match!(t::Parsers.Trie, io::IO, r::Parsers.Result, setvalue::Bool=true, ignorecase::Bool=false)
-
-    Function that takes an `io::IO` argument, a prebuilt `r::Parsers.Result` argument, and a `t::Parsers.Trie` argument, and attempts to match/detect special values in `t` with the next bytes consumed from `io`.
-    If special values are found, `r.result` will be set to the value that was associated with `t` when it was constructed.
-    The return value of `Parsers.match!` is if a special value was indeed detected in `io` (`true` or `false`).
-    Optionally, if the `setvalue` is `false`, `r.result` will be unaffected (i.e. not set) even if a special value is found.
-    The optional argument `ignorecase` can be used if case-insensitive matching is desired.
-
-    Note that `io` is reset to its original position if no special value is found.
-"""
-function match! end
```

Comment:
Can we leave the docs + generic function definition at the top here please? That makes it a bit easier to find if someone opens up the file and shouldn't matter for the underlying issue here.

Reply:
@quinnj I moved them back at the top